src/utils.rs

```diff
@@ -670,6 +670,52 @@ impl Drop for FlushToZeroDenormalsAreZeroFlags {
     }
 }
 
+// This is an RAII structure that disables floating point exceptions while
+// it is alive, so that operations which generate NaNs and infinite values
+// intentionally will not trip an exception when debugging problematic
+// code that is generating NaNs and infinite values erroneously.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub(crate) struct DisableFloatingPointExceptionsFlags {
+    #[cfg(feature = "avoid-fe-exceptions")]
+    // We can't get a precise size for this, because it's of type
+    // `fenv_t`, which is a definition that doesn't exist in rust
+    // (not even in the libc crate, as of the time of writing.)
+    // But since the state is intended to be stored on the stack,
+    // 256 bytes should be more than enough.
+    original_flags: [u8; 256],
```

Comment:
Unfortunately, the code I'd hoped to use which simply checked the status of the exception flags doesn't seem to be possible; there isn't a function for just retrieving the state of the flags into a `c_int`. The only way to actually retrieve and save the state is to store it inside of a `fenv_t`, which is a structure definition which is platform-dependent and could theoretically have an arbitrary size. The functions `fegetexceptflag` and `fetestexcept` appear to refer to the *state* of the floating-point exceptions (i.e., whether they have been raised), and *not* whether or not those exceptions are *enabled*.

So, this code is definitely not ideal for it, since if you enable it on a platform where a `fenv_t` is larger than 256 bytes, it will just explode your stack. But since this feature flag is only meant for debugging I feel like that's okay for now, even if I hate it.